From 4e4550bfc79616306c289dd0030a70ec9e0f8a33 Mon Sep 17 00:00:00 2001
From: Xiaofei Ma <xiaofei@motorola.com>
Date: Fri, 25 Mar 2016 21:12:54 -0500
Subject: [PATCH 831/959] IKFPS-686 input: synaptics_dsx_i2c: touch exclusion
 for fps

Clipa is still null when clipping is turned
on by FPS module.

Change-Id: If20b8cd7288f1b142027e72be9a54a532e9242c0
Signed-off-by: Xiaofei Ma <xiaofei@motorola.com>
Reviewed-on: http://gerrit.mot.com/837763
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Konstantin Makariev <kmakariev@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/input/touchscreen/synaptics_dsx_i2c.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/input/touchscreen/synaptics_dsx_i2c.c b/drivers/input/touchscreen/synaptics_dsx_i2c.c
index e1b3781b2fc..f38f37fc84b 100644
--- a/drivers/input/touchscreen/synaptics_dsx_i2c.c
+++ b/drivers/input/touchscreen/synaptics_dsx_i2c.c
@@ -7256,6 +7256,14 @@ static int fps_notifier_callback(struct notifier_block *self,
 
 	if (rmi4_data && event == 0xBEEF &&
 			rmi4_data && rmi4_data->i2c_client) {
+		struct config_modifier *cm =
+			modifier_by_id(rmi4_data, SYNA_MOD_FPS);
+		if (!cm) {
+			dev_err(&rmi4_data->i2c_client->dev,
+				"No FPS modifier found\n");
+			goto done;
+		}
+
 		state = synaptics_dsx_get_state_safe(rmi4_data);
 		dev_dbg(&rmi4_data->i2c_client->dev,
 			"FPS: %s(%d), suspend flag: %d, BL flag: %d\n",
@@ -7264,12 +7272,17 @@ static int fps_notifier_callback(struct notifier_block *self,
 			rmi4_data->in_bootloader);
 		if (fps_state) {/* on */
 			rmi4_data->clipping_on = true;
+			rmi4_data->clipa = cm->clipa;
+			cm->effective = true;
 		} else {/* off */
 			rmi4_data->clipping_on = false;
+			rmi4_data->clipa = NULL;
+			cm->effective = false;
 		}
 		pr_info("FPS: clipping is %s\n",
 			rmi4_data->clipping_on ? "ON" : "OFF");
 	}
+done:
 	return 0;
 }
 
-- 
2.11.0

